engine/src/test/scala/cromwell/engine/workflow/lifecycle/execution/job/preparation/JobPreparationActorSpec.scala

```diff
@@ -158,26 +159,31 @@ class JobPreparationActorSpec
     }
   }
 
-  it should "retry with more memory attribute when `memoryMultiplier` in BackendJobDescriptorKey is greater than 1" in {
-    val attributes = Map(
-      "memory" -> WomString("2 GB")
+  it should "lookup MemoryMultiplier key/value if available and accordingly update runtime attributes" in {
+    val attributes = Map (
+      "memory" -> WomString("1.1 GB")
     )
     val inputsAndAttributes = (inputs, attributes).validNel
-
-    val actor = TestActorRef(helper.buildTestJobPreparationActor(
-      backpressureTimeout = null,
-      noResponseTimeout = null,
-      dockerHashCredentials = null,
-      inputsAndAttributes = inputsAndAttributes,
-      kvStoreKeysForPrefetch = List.empty,
-      jobKey = helper.mockJobKeyWithMemoryMultiplier
-    ), self)
+    val prefetchedKey = "MemoryMultiplier"
+    val prefetchedVal = KvPair(helper.scopedKeyMaker(prefetchedKey), "1.1")
+    val prefetchedValues = Map(prefetchedKey -> prefetchedVal)
+    var keysToPrefetch = List(prefetchedKey)
+    val actor = TestActorRef(helper.buildTestJobPreparationActor(1 minute, 1 minutes, List.empty, inputsAndAttributes, List(prefetchedKey)), self)
     actor ! Start(ValueStore.empty)
+
+    def respondFromKv(): Unit = {
+      helper.serviceRegistryProbe.expectMsgPF(max = 100 milliseconds) {
+        case KvGet(k) if keysToPrefetch.contains(k.key) =>
+          actor.tell(msg = prefetchedValues(k.key), sender = helper.serviceRegistryProbe.ref)
+          keysToPrefetch = keysToPrefetch diff List(k.key)
+      }
+    }
+    respondFromKv()
```

Comment:
Is there a reason why you made this a function rather than evaluating it inline?

Reply:
I am not sure why I put this as a function. Maybe it was used in 2 places before. But I have updated it to be evaluated inline.